language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToConstantCodeAction.java

```diff
@@ -171,15 +181,29 @@ private String getConstantName(CodeActionContext context) {
         return NameUtil.generateTypeName(CONSTANT_NAME_PREFIX, allNames);
     }
 
-    private Position getPosition(CodeActionContext context) {
+    private ConstantData getConstantData(CodeActionContext context) {
         ModulePartNode modulePartNode = context.currentSyntaxTree().get().rootNode();
         NodeList<ImportDeclarationNode> importsList = modulePartNode.imports();
-        
+
+        ConstantDeclarationNode declarationNode = null;
+        for (Node node : modulePartNode.children()) {
+            if (node.kind() == SyntaxKind.CONST_DECLARATION) {
+                declarationNode = (ConstantDeclarationNode) node;
+                break;
+            }
+        }
+
         if (importsList.isEmpty()) {
-            return PositionUtil.toPosition(modulePartNode.lineRange().startLine());
+            return new ConstantData(
+                    PositionUtil.toPosition(modulePartNode.lineRange().startLine()), false);
+        } else if (declarationNode == null) {
+            ImportDeclarationNode lastImport = importsList.get(importsList.size() - 1);
+            return new ConstantData(
+                    new Position(lastImport.lineRange().endLine().line() + 1, 0), true);
+        } else {
+            return new ConstantData(new Position(declarationNode.lineRange().startLine().line(), 0),
+                    !declarationNode.toString().startsWith("\n"));
```

Comment:
Seems like we don't have a test case for this scenario. Shall we add one?

Also, shall we use `System.lineSeparator()` instead of hardcoding `\n` here? That can change on windows.